tests/functionality/pflow/selfcheck.cpp

```diff
@@ -39,28 +39,38 @@ struct PflowFunctionalityTests
                           int logging_verbosity = EXAGO_LOG_INFO)
       : FunctionalityTestContext(testsuite_filename, logging_verbosity),
         comm{comm} {
+    int my_rank;
+    auto rerr = MPI_Comm_rank(comm, &my_rank);
+    if (rerr != MPI_SUCCESS)
+      throw ExaGOError("Error getting MPI rank number");
+
     auto err = MPI_Comm_size(comm, &nprocs);
-    if (err)
+    if (err != MPI_SUCCESS)
       throw ExaGOError("Error getting MPI num ranks");
   }
 
   void
   ensure_options_are_consistent(toml::value testcase,
                                 toml::value presets = toml::value{}) override {
 
+    int my_rank;
+    auto err = MPI_Comm_rank(comm, &my_rank);
+    if (err != MPI_SUCCESS)
+      throw ExaGOError("Error getting MPI rank number");
```

Comment:
We get rank more than one time here. While not the biggest issue, can you think of a way to only perform this once and to keep this in perhaps the testing initialization routines instead of within these lower level functions?

Since we have an abstract test base, that could also be the place for this